google/cloud/bigtable/internal/rate_limiter.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/bigtable/internal/rate_limiter.h"
+#include "absl/time/time.h"
+
+namespace google {
+namespace cloud {
+namespace bigtable_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+RateLimiter::Clock::duration RateLimiter::acquire(std::int64_t permits) {
+  auto const now = clock_->Now();
+  std::lock_guard<std::mutex> lk(mu_);
+  if (next_ <= now) {
```

Comment:
```suggestion
  if (next_ > now) {
    auto wait = next_ - now;
    next_ += permits * period_;
    return wait;
  }
```

Reply:
moot now, but `if (next > now)` was the simpler of the branches. Thanks.